dvc/dependency/repo.py

```diff
@@ -46,14 +47,33 @@ def _make_repo(self, **overrides):
         with external_repo(**merge(self.def_repo, overrides)) as repo:
             yield repo
 
-    def status(self):
-        with self._make_repo() as repo:
-            current = repo.find_out_by_relpath(self.def_path).info
+    def _get_checksum(self, updated=False):
+        rev_lock = None
+        if not updated:
+            rev_lock = self.def_repo.get(self.PARAM_REV_LOCK)
 
-        with self._make_repo(rev_lock=None) as repo:
-            updated = repo.find_out_by_relpath(self.def_path).info
+        try:
+            with self._make_repo(rev_lock=rev_lock) as repo:
+                try:
+                    return repo.find_out_by_relpath(self.def_path).info["md5"]
+                except OutputNotFoundError:
+                    # Fall through to after exception handler
+                    path = os.path.join(repo.root_dir, self.def_path)
+
+        except NotDvcRepoError:
+            repo_path = cached_clone(
+                self.def_repo[self.PARAM_URL],
+                rev=rev_lock or self.def_repo.get(self.PARAM_REV),
+            )
+            path = os.path.join(repo_path, self.def_path)
+
```

Comment:
@fabiosantoscode Please take a look here https://github.com/iterative/dvc/blob/master/dvc/repo/get.py#L66 . You can do pretty much the same and won't have to duplicate logic or catch raw OutputNotFoundError